src/pages/iou/IOUTransactions.js

```diff
@@ -18,33 +18,80 @@ const propTypes = {
 
     /** ReportID for the associated IOU report */
     iouReportID: PropTypes.number.isRequired,
+
+    /** Email for the authenticated user */
+    userEmail: PropTypes.string.isRequired,
+
+    /** Has the iouReport been paid? */
+    isIOUReportPaid: PropTypes.bool,
 };
 
 const defaultProps = {
     reportActions: {},
+    isIOUReportPaid: false,
 };
 
-const IOUTransactions = ({
-    reportActions,
-    chatReportID,
-    iouReportID,
-}) => (
-    <View style={[styles.mt3]}>
-        {_.map(reportActions, (reportAction) => {
-            if (reportAction.actionName === CONST.REPORT.ACTIONS.TYPE.IOU
-                && reportAction.originalMessage.IOUReportID === iouReportID) {
-                return (
-                    <ReportTransaction
-                        chatReportID={chatReportID}
-                        iouReportID={iouReportID}
-                        action={reportAction}
-                        key={reportAction.sequenceNumber}
-                    />
-                );
-            }
-        })}
-    </View>
-);
+class IOUTransactions extends Component {
+    constructor(props) {
+        super(props);
+
+        this.getRejectableTransactions = this.getRejectableTransactions.bind(this);
+    }
+
+    /**
+     * Builds and returns the rejectableTransactionIDs array. A transaction must meet multiple requirements in order
+     * to be rejectable. We must exclude transactions not associated with the iouReportID, actions which have already
+     * been rejected, and those which are not of type 'create'.
+     *
+     * @returns {Array}
+     */
+    getRejectableTransactions() {
+        if (this.props.isIOUReportPaid) {
+            return [];
+        }
+
+        const actionsForIOUReport = _.filter(this.props.reportActions, action => action.originalMessage
+            && action.originalMessage.type && action.originalMessage.IOUReportID === this.props.iouReportID);
+
+        const rejectedTransactionIDs = _.chain(actionsForIOUReport)
+            .filter(action => ['cancel', 'decline'].includes(action.originalMessage.type))
+            .map(rejectedAction => lodashGet(rejectedAction, 'originalMessage.IOUTransactionID', ''))
```

Comment:
Since this is defaulting to `''` I think you'll also want to add a `.flatten()` to the chain (same as below). Either that, or don't use a default and just trust that it's not necessary.